src/client/mod.rs

```diff
@@ -71,21 +71,12 @@ static CLIENT_REQUEST_DURATION: LazyLock<CounterVec> = LazyLock::new(|| {
 type HmacSha512 = Hmac<Sha512>;
 
 // -----------------------------------------------------------------------------
-// Request trait
+// Execute trait
 
-pub trait Request {
+/// Execute HTTP requests.
+pub trait Execute {
     type Error;
 
-    fn request<T, U>(
```

Comment:
You are breaking the abi with this modification.

Reply:
Yes I'd like to propose this breaking change to API because IMHO `request` is better placed in the REST interface, as it's a shorthand for the implementation of the `Method` specific methods